src/lib/math.js

```diff
@@ -326,8 +326,40 @@ const $builtinmodule = function (name) {
         }
     };
 
-    function isqrt(x) {
-        throw new Sk.builtin.NotImplementedError("math.isqrt() is not yet implemented in Skulpt");
+    function bit_length(i) {
+        return new Sk.builtin.int_(Sk.builtin.bin(i).sq$length() - 2);
+    }
+
+    function isqrt(n) {
+        n_js = Sk.misceval.asIndexOrThrow(n);
+        n = new Sk.builtin.int_(n_js);
+        if (n_js < 0) {
+            throw new Sk.builtin.ValueError("isqrt() argument must be nonnegative");
+        }
+        if (n_js === 0) {
+            return new Sk.builtin.int_(0);
+        }
+        const one = new Sk.builtin.int_(1);
+        const two = new Sk.builtin.int_(2);
+        let c = bit_length(n);
+        c = c.nb$subtract(one).nb$floor_divide(two);
+
+        let a = new Sk.builtin.int_(1);
+        let d = new Sk.builtin.int_(0);
+        let s;
+        if (Sk.ffi.remapToJs(c) > 0) {
```

Comment:
I needed this because the following gives different values in Python and in Skulpt
```
i = 0
print(i.bit_length())
```
Returns 0 in Python
Returns 1 in Skulpt

Reply:
Fixed in #1362